runtime/onert/api/nnfw/src/nnfw_session.cc

```diff
@@ -113,11 +113,12 @@ std::string trim(std::string_view value)
   return std::string(value.substr(begin, end - begin + 1));
 }
 
-std::string inferModelType(const std::string &filename)
+std::string inferModelType(const std::filesystem::path &file_path)
 {
-  std::filesystem::path file_path(filename);
-  std::string ext = file_path.extension().string();
-  return ext.empty() ? "" : ext.substr(1);
+  if (!file_path.has_extension())
+    return "";
+
+  return file_path.extension().string().substr(1);
```

Comment:
The `loadModel()` function checks model type in a case-sensitive way. But file name extensions in most cases are treated in a cases-insensitive way, e.g. photo.jpg, photo.JPG, or config.yaml, config.YAML - these files are recognized as JPEG and YAML by e.g. VSCode. How about converting the extension to lowercase here?